pkg/inventory/inventory-client.go

```diff
@@ -113,24 +124,32 @@ func (cic *ClusterInventoryClient) Merge(localInv InventoryInfo, objs object.Obj
 		if err := cic.createInventoryObj(invInfo, dryRun); err != nil {
 			return nil, err
 		}
+		if err := cic.updateStatus(invInfo, dryRun); err != nil {
+			return nil, err
+		}
 	} else {
 		// Update existing cluster inventory with merged union of objects
 		clusterObjs, err := cic.GetClusterObjs(localInv, dryRun)
 		if err != nil {
 			return pruneIds, err
 		}
-		if objs.Equal(clusterObjs) {
-			klog.V(4).Infof("applied objects same as cluster inventory: do nothing")
-			return pruneIds, nil
-		}
 		pruneIds = clusterObjs.Diff(objs)
 		unionObjs := clusterObjs.Union(objs)
+		applyStatus := getApplyStatus(pruneIds, unionObjs)
 		klog.V(4).Infof("num objects to prune: %d", len(pruneIds))
 		klog.V(4).Infof("num merged objects to store in inventory: %d", len(unionObjs))
 		wrappedInv := cic.InventoryFactoryFunc(clusterInv)
-		if err = wrappedInv.Store(unionObjs); err != nil {
+		if err = wrappedInv.Store(unionObjs, applyStatus); err != nil {
 			return pruneIds, err
 		}
+		clusterInv, err := wrappedInv.GetObject()
+		if objs.Equal(clusterObjs) {
+			klog.V(4).Infof("applied objects same as cluster inventory: do nothing")
```

Comment:
This log isn't accurate any more if the status is still being updated.

Reply:
Removed this log as inverting the if statement.